pkg/drain/nodeDrainStrategy.go

```diff
@@ -49,18 +52,27 @@ func (ds *osdDrainStrategy) Execute(node *corev1.Node, logger logr.Logger) ([]*D
 			return nil, fmt.Errorf("cannot determine drain commencement time for node %v", node.Name)
 		}
 		me := &multierror.Error{}
-		for _, ds := range ds.timedDrainStrategies {
-			dsName := ds.GetName()
-			expectedTime := result.AddedAt.Add(ds.GetWaitDuration())
+		for _, tds := range ds.timedDrainStrategies {
+			dsName := tds.GetName()
+			expectedTime := result.AddedAt.Add(tds.GetWaitDuration())
 			drainStrategyMsg := fmt.Sprintf("drain strategy %v for node %v, commencing drain at %v, execution expected after %v", dsName, node.Name, result.AddedAt, expectedTime)
-			if isAfter(result.AddedAt, ds.GetWaitDuration()) {
+			if isAfter(result.AddedAt, tds.GetWaitDuration()) {
 				logger.Info(fmt.Sprintf("Executing %s", drainStrategyMsg))
-				r, err := ds.GetStrategy().Execute(node, logger)
+				r, err := tds.GetStrategy().Execute(node, logger)
 				if err != nil {
 					return nil, err
 				}
 				me = multierror.Append(err, me)
 				if r.HasExecuted {
+					if dsName == pdbPodDeleteName {
+						logger.Info("Sending upgrade delay message about node drain grace period")
+						msg := "Node drain grace period might be impacting cluster upgrade. " +
+							"Please refer to the article for further details https://access.redhat.com/solutions/7075425"
+						err = ds.notifier.NotifyState(notifier.MuoStateDelayed, msg)
```

Comment:
Here the notification is expected to be sent multiple times when the drain strategy is run which will be multiple notifications for the customer. We want to control it.

Reply:
Have added logic to check metrics before the service log is sent.